futures-channel/src/lib.rs

```diff
@@ -9,6 +9,8 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 #![warn(missing_docs, missing_debug_implementations, rust_2018_idioms, unreachable_pub)]
+// It cannot be included in the published code because this lints have false positives in the minimum required version.
+#![cfg_attr(test, warn(single_use_lifetimes))]
```

Comment:
This enables `single_use_lifetimes` lint when running `cargo test`, `cargo check --tests` etc. (FWIW, I often run `cargo check --all-features --tests --all` locally before opening PR).